### ISSUE TITLE
HTML errors

### ISSUE THREAD

**Issue by garyemiller** [cyoung/stratux#630]
https://validator.w3.org/ finds html warnings and errors in /var/www/plates.index.html

See this screenshot of validator output:

![screenshot-validator w3 org-2017-06-27-15-53-25](https://user-images.githubusercontent.com/1270340/27613528-e3b9f42e-5b50-11e7-9ce8-837416219541.png)

**Comment by bradanlane** [comment]
It looks like that validator is not aware of several things including:

- Stratus Admin Web UI is a mobile web application
- JavaScript frameworks add their own attributes to HTML elements
- some validator warnings are actually in conflict with W3 media-query specifications

A few of the validator messages suggest an opportunity to clean up some HTML/CSS as the specification has evolved over the past 2 years. At the same time, not all browsers are up-to-date on the specification so changes may do as much harm as good.

**Comment by JohnOCFII** [comment]
It looks like the HTML evaluator hasn't been updated in about 4 years either.

**Comment by bradanlane** [comment]
Good catch John. That would explain both the erroneous "viewport" messages and the "media query" messages. It may also be why the validator is flagging all of the framework attributes on HTML elements.

If this is an important issue, a better validator should be the benchmark.

**Comment by garyemiller** [comment]
There is no such thing as 'mobile html' there is only 'valid html.  The way the Stratux JS is done is counter to the HTML specification.

**Comment by bradanlane** [comment]
Correct. A "mobile web app" is written with standard HTML, CSS, and optionally JavaScript. 

The "Stratux JS" uses an open source industry JavaScript framework. The framework was created and open sourced by Google.

**Comment by garyemiller** [comment]
'industry standard'?  Can you point me to an RFC, or other similar standard?

I am familiar with the 'HTML Standard', provided by the w3c.org.  I'd be happy to point out specifics how Startux conflicts with the published international HTML Standard.

**Comment by bradanlane** [comment]
Just like I never stated "mobile HTML", I didn't state "industry standard".

**Comment by garyemiller** [comment]
So we agree the Stratux code does not comply with the HTML industry standard?

**Comment by bradanlane** [comment]
Agree to disagree :-)

"Industry standards ... the generally accepted requirements followed by the members of an industry."

**Comment by garyemiller** [comment]
So w3c.org is not the consensus of the members of the industry?  Must be fun down in your rabbit hole.

If you don't care to get it right, then I'm not gonna...

**Comment by bradanlane** [comment]
The standards bodies are comprised of industry leaders and thus there is a lag between industry advancements and standards bodies.

**Comment by garyemiller** [comment]
And the stratux html  appears to lag behind even more.  It still has XHTML cruft that was removed years ago from the HTML standard.

**Comment by JohnOCFII** [comment]
I'm sure a working Pull Request would be graciously accepted.

**Comment by bradanlane** [comment]
(consolidated prior comments)

"Industry standards ... the generally accepted requirements followed by the members of an industry."

_The validator website generates validation errors._ :-)

If contributors feel these validation messages are important, fork the project, make the corrections, and submit a pull request.

**Comment by garyemiller** [comment]
"The validator website generates validation errors. :-)"

Uh, no:  https://validator.w3.org/check?uri=https%3A%2F%2Fvalidator.w3.org&charset=%28detect+automatically%29&doctype=Inline&group=0&verbose=1&user-agent=W3C_Validator%2F1.3+http%3A%2F%2Fvalidator.w3.org%2Fservices

**Comment by bradanlane** [comment]
I stand corrected. I pasted the wrong URL. I pasted the github URL.

**Comment by Ergonomicmike** [comment]
I checked a branched settings.html on both checkers (including the /nu) and they both report errors. I don't mind noodling around to try to reduce the reported errors. But to do that, I need to know what doctype the Stratux wants to use. (And this might already be declared somewhere ... I seem to recall that there was a master webpage for the Stratux UI, and then html was swapped in and out of it.)

**Comment by Ergonomicmike** [comment]
So, the first place to start cleaning up - if, in fact, it is necessary, and if the team wants to do it - is with the index.html page. For if it's not right, none of the sub pages that script into it will be either. I remember a year or two ago that the appcache thing was done for good reason. I don't know anything about 'service workers.' This sounds like it might be more than just HTML clean up.

On some of the other error reports, it seems like maybe it's as simple as hyphens aren't allowed in div labels? Even if it's as simple as that, this could become an quite a project, trying to find and retie loose ends.
